packages/core/src/context/MemoryStore.ts

```
@@ -0,0 +1,443 @@
+  lastIndexed?: Date;
+  error?: string;
+}
+
+/**
+ * MemoryStore service that extends existing memory capabilities with semantic search.
+ * Works alongside MemoryTool to provide vector search over stored memories.
+ */
+export class MemoryStore {
+  private vectorService: VectorIndexService | null = null;
+  private contentGenerator: ContentGenerator;
+  private collectionName: string = 'memory_embeddings';
```

Biome (1.9.4)

[error] This type annotation is trivially inferred from its initialization.

Safe fix: Remove the type annotation.


(lint/style/noInferrableTypes)